src/frontend/src/components/RecordSearch/index.tsx

```diff
@@ -2,14 +2,103 @@ import React from 'react';
 import { connect } from "react-redux";
 import { AppState } from "../../redux/store";
 import { loadRecords } from "../../redux/records/actions";
-
-// This is a placeholder component. I created it to prototype
-// out the general structure we'll need when working with
-// Records using Redux.
+import moment from 'moment';
 
 class RecordSearch extends React.Component {
+  state = {
+    firstName: null,
+    lastName: null,
+    dateOfBirth: '', // Moment expects a string to be passed in as a paramenter in the validateForm function.
+    missingFirstName: undefined, // Undefined is used here to prevent aria-invalid from rendering until form submission.
+    missingLastName: undefined,
+    missingInputs: null,
+    invalidDate: undefined
+  }
+
+  handleChange = (e: React.BaseSyntheticEvent) => {
+    this.setState({
+      [e.target.id]: e.target.value
+    });
+  }
+
+  handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+    this.validateForm();
+    if (this.state.missingInputs === false && this.state.invalidDate === false) {
+      // Dispatch an action.
+    };
+  }
+
+  validateForm = () => {
+    !this.state.firstName ? this.setState({missingFirstName: true}) : this.setState({missingFirstName: false});
+    !this.state.lastName ? this.setState({missingLastName: true}) : this.setState({missingLastName: false});
+
+    !this.state.firstName || !this.state.lastName || !this.state.dateOfBirth ?
+    this.setState({missingInputs: true}) : this.setState({missingInputs: false});
+
+    moment(this.state.dateOfBirth,'MM/DD/YYYY',true).isValid() === false ?
+    this.setState({invalidDate: true}) : this.setState({invalidDate: false});
+  }
```

Comment:
All of the statements in this method are of the form:

    [boolean expression] ? do_foo(true) : do_foo(false)

You can rewrite (and thus shorten/simplify them) as:

    do_foo([boolean expression])

Reply:
Good catch! I've rewritten them as suggested.